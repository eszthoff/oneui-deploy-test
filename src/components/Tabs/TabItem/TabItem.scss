.TabItem {
    background-color: var(--color-neutral-25);
    border: {
        color: var(--color-neutral);
        style: solid;
        top-left-radius: var(--border-radius);
        top-right-radius: var(--border-radius);
        width: var(--line-normal);
        bottom-width: 0;
    }
    color: var(--link-color-normal);
    cursor: pointer;
    font: {
        family: var(--font-family-primary);
        size: var(--font-size-normal);
    }
    margin: {
        top: var(--margin);
        right: var(--margin);
        left: var(--margin);
    };
    outline: none;
    padding: var(--spacing-2x) var(--spacing-4x);
    line-height: var(--line-height-normal);
    text-decoration: none;
    transition: border-color var(--transition-duration);

    &:hover {
        border-color: var(--color-neutral-60);
    }

    &--isActive {
        background-color: var(--color-background);
        color: var(--color-foreground);
        margin-bottom: -1px;

        &:hover {
            border-color: var(--color-neutral);
        }
    }
}
