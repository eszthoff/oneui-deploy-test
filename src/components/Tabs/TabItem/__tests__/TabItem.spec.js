import React from 'react';
import toJson from 'enzyme-to-json';
import TabItem from '../TabItem';

describe('<TabItem>', () => {
    it('should render a tab item correctly', () => {
        const wrapper = shallow(<TabItem label="Some tab" />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should add classes for props', () => {
        const wrapper = shallow(<TabItem label="Some tab" isActive />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should allow node type labels', () => {
        const wrapper = shallow(<TabItem label={<strong>Some tab</strong>} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
