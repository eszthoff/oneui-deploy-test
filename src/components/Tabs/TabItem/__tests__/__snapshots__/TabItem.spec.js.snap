// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabItem> should add classes for props 1`] = `
<a
  className="TabItem TabItem--isActive"
  href="#"
>
  Some tab
</a>
`;

exports[`<TabItem> should allow node type labels 1`] = `
<a
  className="TabItem"
  href="#"
>
  <strong>
    Some tab
  </strong>
</a>
`;

exports[`<TabItem> should render a tab item correctly 1`] = `
<a
  className="TabItem"
  href="#"
>
  Some tab
</a>
`;
